DigitalLearningSolutions.Web/Controllers/LearningMenuController/LearningMenuController.cs

```diff
@@ -56,10 +64,37 @@ public IActionResult Index(int customisationId)
             return View(model);
         }
 
+        [Route("/LearningMenu/Close")]
+        public IActionResult Close()
+        {
+            sessionService.StopSession(User.GetCandidateId());
+            HttpContext.Session.Clear();
+
+            return RedirectToAction("Current", "LearningPortal");
+        }
+
         public IActionResult ContentViewer()
         {
             var model = new ContentViewerViewModel(config);
             return View(model);
         }
+
+        private void StartOrUpdateSession(int candidateId, int customisationId)
+        {
+            var currentSessionId = HttpContext.Session.GetInt32($"SessionID-{customisationId}");

```

Comment:
Hm, it would be nice if the complication of interacting with `HttpContext.Session` could also be extracted to the SessionService. What do you think? 🙂

Reply:
Yeah that might be nice. How would I inject the `HttpContext` into the `SessionService`? 🙂

Reply:
I think you'd probably need to pass it (or just the session) to each method. Which is a bit annoying but it's just one extra parameter so shouldn't be too bad. What do you think? 🙂

Reply:
Changed over to this in d29430b 🙂